__tests__/table_moveRow.js

```diff
@@ -0,0 +1,489 @@
+        doc(
+          table(
+            row(td(p('3')), tdEmpty, tdEmpty),
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('2')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+    });
+
+    it('should move row left-to-right', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(td(p('1')), tdEmpty, tdEmpty),
+            row(td(p('2')), td(p('x')), tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 2)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('1')), tdEmpty, tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty),
+            row(td(p('2')), td(p('x')), tdEmpty)
+          )
+        )
+      );
+    });
+  });
+
+  describe('on a table with merged cells', () => {
+    it('should move columns merged at first line', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(td(p('1')), td({ colspan: 2 }, p('merged cell'))),
+            row(td(p('2')), tdCursor, tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('2')), tdCursor, tdEmpty),
+            row(td(p('1')), td({ colspan: 2 }, p('merged cell'))),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+    });
+
+    it('should move columns merged at middle line', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('2')), td({ colspan: 2 }, p('merged cell'))),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('2')), td({ colspan: 2 }, p('merged cell'))),
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+    });
+
+    it('should move columns merged at last line', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('2')), tdEmpty, tdEmpty),
+            row(td(p('3')), td({ colspan: 2 }, p('merged cell')))
+          )
+        )
+      );
+
+      const newTr = moveRow(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('2')), tdEmpty, tdEmpty),
+            row(td(p('1')), tdCursor, tdEmpty),
+            row(td(p('3')), td({ colspan: 2 }, p('merged cell')))
+          )
+        )
+      );
+    });
+
+    it('should move and keep table headers', () => {
```

Comment:
do we have the same behaviour for header columns? would you like to add a test for it?

Reply:
Good point, let me write tests for it

- [ ] Write tests for header columns